packages/entity-server/src/routes/hierarchy/middleware/fields.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { reduceToDictionary, reduceToArrayDictionary } from '@tupaia/utils';
+import { EntityType, EntityFields } from '../../../models';
+import {
+  HierarchyRequest,
+  HierarchyContext,
+  ExtendedEntityFields,
+  EntityResponseObject,
+} from '../types';
+import {
+  extendedFieldFunctions,
+  getParentCodeBulk,
+  getChildrenCodesBulk,
+} from '../extendedFieldFunctions';
+
+const validFields: (keyof EntityFields)[] = ['code', 'country_code'];
```

Comment:
@rohan-bes - is there any reason why `name` `image_url` and `type` fields are not included as valid fields? I also checked in ExtendedEntityFields and didn't see them.

Reply:
Those are coming in the second PR: https://github.com/beyondessential/tupaia/pull/2266
👍